src/utils/transaction/insertTransaction.ts

```diff
@@ -15,41 +16,134 @@ import { reportUsage } from "../usage";
+ * 4. Addresses must be normalized to checksum format
+ * 5. Properties like accountSignerAddress, accountFactoryAddress, and entrypoint
+ *    are only available on SmartBackendWalletDetails type
+ * 6. Only smart backend wallets can be found via accountAddress lookup when 'from'
+ *    is not found - finding anything else indicates invalid wallet configuration
+ */
+
 interface InsertTransactionData {
   insertedTransaction: InsertedTransaction;
   idempotencyKey?: string;
   shouldSimulate?: boolean;
 }
 
+const validateSmartBackendWalletChainSupport = async (chainId: number) => {
+  if (!(await doesChainSupportService(chainId, "account-abstraction"))) {
+    throw createCustomError(
+      "Chain does not support smart backend wallets",
+      StatusCodes.BAD_REQUEST,
+      "SBW_CHAIN_NOT_SUPPORTED",
+    );
+  }
+};
+
+const transformV5SmartBackendWallet = async (
+  transaction: QueuedTransaction,
+  walletDetails: SmartBackendWalletDetails,
+): Promise<QueuedTransaction> => {
+  await validateSmartBackendWalletChainSupport(transaction.chainId);
+
+  if (transaction.accountAddress) {
+    throw createCustomError(
+      "Smart backend wallets do not support interacting with other smart accounts",
+      StatusCodes.BAD_REQUEST,
+      "INVALID_SMART_BACKEND_WALLET_INTERACTION",
+    );
+  }
+
+  return {
+    ...transaction,
+    isUserOp: true,
+    signerAddress: walletDetails.accountSignerAddress,
+    from: walletDetails.accountSignerAddress,
+    accountAddress: transaction.from,
+    target: transaction.to,
+    accountFactoryAddress: walletDetails.accountFactoryAddress ?? undefined,
+    entrypointAddress: walletDetails.entrypointAddress ?? undefined,
+    walletType: walletDetails.type,
+  };
+};
+
+const transformV4SmartBackendWallet = async (
+  transaction: QueuedTransaction,
+  walletDetails: SmartBackendWalletDetails,
+): Promise<QueuedTransaction> => {
+  await validateSmartBackendWalletChainSupport(transaction.chainId);
+
+  return {
+    ...transaction,
+    entrypointAddress: walletDetails.entrypointAddress ?? undefined,
+    accountFactoryAddress: walletDetails.accountFactoryAddress ?? undefined,
+    walletType: walletDetails.type,
+  };
+};
+
 /**
  * Enqueue a transaction to be submitted onchain.
- *
- * @param args
- * @returns queueId
  */
 export const insertTransaction = async (
   args: InsertTransactionData,
 ): Promise<string> => {
   const { insertedTransaction, idempotencyKey, shouldSimulate = false } = args;
 
-  // The queueId uniquely represents an enqueued transaction.
-  // It's also used as the idempotency key (default = no idempotence).
-  let queueId: string = randomUUID();
-  if (idempotencyKey) {
-    queueId = idempotencyKey;
-    if (await TransactionDB.exists(queueId)) {
-      // No-op. Return the existing queueId.
-      return queueId;
-    }
+  // Handle idempotency
+  const queueId: string = idempotencyKey ?? randomUUID();
+  if (idempotencyKey && (await TransactionDB.exists(queueId))) {
```

Comment:
I see you're reducing this code for readability. Actually maybe this line can just be:
```typescript
if (await TransactionDB.exists(queueId)) {
```

The key difference being: if a RANDOMLY GENERATED queueId exists in DB (idempotency was not provided) this will fail. Previously, it would override the previous one. Arguably both are bad situations but extremely low probability, so might as well go with the simpler code.